components/availability/src/Availability.svelte

```diff
@@ -63,17 +60,49 @@
       .slice(0, -1) // dont show the 25th hour
       .map((time) => {
         const endTime = timeMinute.offset(time, slot_size);
-
-        let slotIsAvailable = true; // default
-        if (available_times.length) {
-          slotIsAvailable = available_times.some((slot) => {
-            return time >= slot.start_time && endTime <= slot.end_time;
+        let freeCalendars = [];
+
+        let availability = "available"; // default
+
+        if (calendars.length) {
+          calendars.forEach((c) => {
+            if (c.availability === "busy") {
+              if (
+                !c.timeslots.some(
+                  (slot) => time >= slot.start_time && endTime <= slot.end_time,
+                )
+              ) {
+                freeCalendars.push(c.email_address);
+              }
+            } else if (c.availability === "free" || !c.availability) {
+              // if they pass in a calendar, but don't have availability, assume the timeslots are available.
+              if (
+                c.timeslots.some(
+                  (slot) => time >= slot.start_time && endTime <= slot.end_time,
```

Comment:
nit - We can define something like this to avoid having to repeat this code
```
const availabilityExists = c.timeslots.some((slot) => time >= slot.start_time && endTime <= slot.end_time)
...
if (!availabilityExists) { ... }
...

if (availabilityExists) { ... }
```